pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -353,3 +369,200 @@ func missingKeysofObjectResults(tr *v1beta1.TaskRun, specResults []v1beta1.TaskR
 	}
 	return findMissingKeys(neededKeys, providedKeys)
 }
+
+func validateParamArrayIndex(ctx context.Context, params []v1beta1.Param, spec *v1beta1.TaskSpec) error {
```

Comment:
this block of code looks extremely similar to what was introduced in the pipelinerun reconciler. Can any of it be moved to a common location?

Reply:
I'm not sure if you're referring to `extractParamIndex` here or `extractParamIndexes` in the PR reconciler but either way yes they could be pulled out. I'd create a new package `pkg/params` and put it there in a separate PR